### ISSUE TITLE
package doesn't install on MacOS with Xcode 10

### ISSUE THREAD

**Issue by jdavies-st** [spacetelescope/spherical_geometry#175]
On Mac OS, `stdlibc++` has been deprecated in Xcode for a few years, since Xcode 8.0, and those libraries are finally removed in Xcode 10.0.  So when one tries to build this package on a system with Xcode 10, it can't find the GNU C++ includes:
```
    ERROR: Complete output from command /Users/jdavies/anaconda3/envs/jwst/bin/python -c 'import setuptools, tokenize;__file__='"'"'/Users/jdavies/dev/spherical_geometry/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps:
    ERROR: running develop
    running egg_info
    creating spherical_geometry.egg-info
    writing spherical_geometry.egg-info/PKG-INFO
    writing dependency_links to spherical_geometry.egg-info/dependency_links.txt
    writing requirements to spherical_geometry.egg-info/requires.txt
    writing top-level names to spherical_geometry.egg-info/top_level.txt
    writing manifest file 'spherical_geometry.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching 'CHANGES.rst'
    warning: no files found matching '*.c' under directory '*.pyx'
    warning: no files found matching '*.pxd' under directory '*.pyx'
    warning: no files found matching '*' under directory 'scripts'
    no previously-included directories found matching 'build'
    no previously-included directories found matching 'docs/_build'
    no previously-included directories found matching 'docs/api'
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '*.o' found anywhere in distribution
    writing manifest file 'spherical_geometry.egg-info/SOURCES.txt'
    running build_ext
    building 'spherical_geometry.math_util' extension
    creating build
    creating build/temp.macosx-10.7-x86_64-3.7
    creating build/temp.macosx-10.7-x86_64-3.7/src
    creating build/temp.macosx-10.7-x86_64-3.7/cextern
    creating build/temp.macosx-10.7-x86_64-3.7/cextern/qd-library
    creating build/temp.macosx-10.7-x86_64-3.7/cextern/qd-library/src
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/jdavies/anaconda3/envs/jwst/include -arch x86_64 -I/Users/jdavies/anaconda3/envs/jwst/include -arch x86_64 -I/private/var/folders/xx/r4myzwyn4z18p6b6dzss2vv40000gq/T/pip-build-env-3npf73nk/overlay/lib/python3.7/site-packages/numpy/core/include -Icextern/qd-library/include -Isrc -I/Users/jdavies/anaconda3/envs/jwst/include/python3.7m -c src/math_util.c -o build/temp.macosx-10.7-x86_64-3.7/src/math_util.o
    In file included from src/math_util.c:2:
    In file included from /private/var/folders/xx/r4myzwyn4z18p6b6dzss2vv40000gq/T/pip-build-env-3npf73nk/overlay/lib/python3.7/site-packages/numpy/core/include/numpy/arrayobject.h:4:
    In file included from /private/var/folders/xx/r4myzwyn4z18p6b6dzss2vv40000gq/T/pip-build-env-3npf73nk/overlay/lib/python3.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:12:
    In file included from /private/var/folders/xx/r4myzwyn4z18p6b6dzss2vv40000gq/T/pip-build-env-3npf73nk/overlay/lib/python3.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1822:
    /private/var/folders/xx/r4myzwyn4z18p6b6dzss2vv40000gq/T/pip-build-env-3npf73nk/overlay/lib/python3.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API, disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
    #warning "Using deprecated NumPy API, disable it with " \
     ^
    src/math_util.c:67:1: warning: unused function 'load_point' [-Wunused-function]
    load_point(const char *in, const intp s, double* out) {
    ^
    2 warnings generated.
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/jdavies/anaconda3/envs/jwst/include -arch x86_64 -I/Users/jdavies/anaconda3/envs/jwst/include -arch x86_64 -I/private/var/folders/xx/r4myzwyn4z18p6b6dzss2vv40000gq/T/pip-build-env-3npf73nk/overlay/lib/python3.7/site-packages/numpy/core/include -Icextern/qd-library/include -Isrc -I/Users/jdavies/anaconda3/envs/jwst/include/python3.7m -c cextern/qd-library/src/bits.cpp -o build/temp.macosx-10.7-x86_64-3.7/cextern/qd-library/src/bits.o
    warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
    cextern/qd-library/src/bits.cpp:14:10: fatal error: 'iostream' file not found
    #include <iostream>
             ^~~~~~~~~~
    1 warning and 1 error generated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command "/Users/jdavies/anaconda3/envs/jwst/bin/python -c 'import setuptools, tokenize;__file__='"'"'/Users/jdavies/dev/spherical_geometry/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps" failed with error code 1 in /Users/jdavies/dev/spherical_geometry/
```
The `setup.py` file needs to be updated to use the correct C++ libraries for Mac OS.

**Comment by jhunkeler** [comment]
I think the problem runs a little deeper. Another one of the deprecations removes the ability to compile C++ with `gcc` (you need `g++`). Forcing it to compile with `g++` throws the opposite error.

I'll look into compiling the `qd` library beforehand, then linking `math_util` to it, rather than attempting to lump the two together into a single library.

Thanks 🍎

**Comment by jdavies-st** [comment]
The following helped me in my local repo:
```
$ git diff
diff --git a/setup.py b/setup.py
index 9b112d2..db9f281 100644
--- a/setup.py
+++ b/setup.py
@@ -3,6 +3,7 @@
 
 import os
 import sys
+import platform
 
 from setuptools import setup
 from setuptools import Extension
@@ -73,6 +74,12 @@ ext_info['include_dirs'].extend([
     qd_library_include_path,
     'src'])
 
+compile_extra_args = []
+link_extra_args = []
+if platform.system() == "Darwin":
+    # compile_extra_args = ['-std=c++11', "-mmacosx-version-min=10.9"]
+    link_extra_args = ["-stdlib=libc++", "-mmacosx-version-min=10.9"]
+
 setup(
     name=PACKAGENAME,
     use_scm_version=True,
@@ -106,6 +113,12 @@ setup(
         ]
     },
     ext_modules=[
-        Extension('spherical_geometry.math_util', sources, **ext_info)
+        Extension(
+            'spherical_geometry.math_util',
+            sources,
+            **ext_info,
+            extra_compile_args=link_extra_args,
+            extra_link_args=link_extra_args,
+        )
     ],
 )
```

**Comment by jdavies-st** [comment]
Closed by #182
